src/config.rs

```diff
@@ -25,21 +25,43 @@ use crate::Result;
 
 /// The configuration for nitrocli, usually read from configuration files and environment
 /// variables.
-#[derive(Clone, Copy, Debug, Default, PartialEq, serde::Deserialize)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, merge::Merge, serde::Deserialize)]
 pub struct Config {
   /// The model to connect to.
   pub model: Option<args::DeviceModel>,
   /// Whether to bypass the cache for all secrets or not.
+  #[merge(strategy = overwrite_if_false)]
   #[serde(default)]
   pub no_cache: bool,
+  #[merge(strategy = overwrite_if_zero)]
   #[serde(default)]
   /// The log level.
   pub verbosity: u8,
 }
 
+fn overwrite_if_false(left: &mut bool, right: &mut bool) {
+  if !*left {
+    *left = *right;
+  }
+}
+
+fn overwrite_if_zero(left: &mut u8, right: &mut u8) {
+  if *left == 0 {
+      *left = *right;
+  }
+}
+
 impl Config {
   pub fn load() -> Result<Self> {
-    load_user_config().map(|o| o.unwrap_or_else(Default::default))
+    use merge::Merge;
+
+    let mut config = Config::default();
+    if let Some(mut user_config) = load_user_config()? {
+        config.merge(&mut user_config);
+    }
+    config.merge(&mut envy::prefixed("NITROCLI_").from_env()?);
```

Comment:
Can you clarify how `envy` will behave if other variables with the `NITROCLI_` prefix are present but not used by the struct? Am I correct to assume that it will just ignore them?

Reply:
Yes.